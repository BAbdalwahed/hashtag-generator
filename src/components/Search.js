import React, { useState } from "react";
import results from "../data/results.json";
import SearchResults from "./SearchResults";

const Search = () => {
  const [searchValue, setSearchValue] = useState("");
  const [searchResults, setSearchResults] = useState("");
  const [loading, setLoading] = useState(true);
  const handleChange = (e) => {
    setSearchValue(e.target.value);
  };
  const handleSubmit = async (e) => {
    e.preventDefault();
    setSearchResults(results[searchValue]);
    await setLoading(false);
  };

  return (
    <div className="search">
      <h1>#Hashtify</h1>
      <div className="search-form">
        <form onSubmit={handleSubmit}>
          {/* Labels have a weird error - let's look into this together */}
          <label htmlFor="search-input">Search for hashtags{}</label>
          <input
            type="text"
            id="search-input"
            name="search-input"
            onChange={handleChange}
          />
          <button type="submit">Search</button>
        </form>
      </div>
      <div className="search-results">
        {!loading && <SearchResults searchResults={searchResults} />}
      </div>
    </div>
  );
};

export default Search;
