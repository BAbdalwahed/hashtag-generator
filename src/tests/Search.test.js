import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Search from "../components/Search";

it("renders Search component", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Search />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
