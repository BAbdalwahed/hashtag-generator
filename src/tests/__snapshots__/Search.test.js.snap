// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Search component 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <h1>
      #Hashtify
    </h1>
    <div
      class="search-form"
    >
      <form>
        <label
          for="search-input"
        >
          Search for hashtags
        </label>
        <input
          id="search-input"
          name="search-input"
          type="text"
        />
        <button
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="search-results"
    />
  </div>
</DocumentFragment>
`;
